### PR TITLE
Enable direct execution of generated script in linux/macos

### PR DESCRIPTION
**Pull request by tspascoal**
Add a `pwsh` shebang to the generated script and if the script is generated on a linux/macos operating system than executable bits are added to the generated file

generated script now starts with
```pwsh
#!/usr/bin/pwsh
....
```

This allows scripts to be executed directly from the CLI like explained in the docs.

Implements #279 

- [x] Did you write/update appropriate tests
- [ ] Release notes updated (if appropriate)
- [ ] Appropriate logging output
- [x] Issue linked

### CODE REVIEW

File: src/gei/gei.csproj [hunk 1]
```diff
@@ -10,6 +10,7 @@
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.11.0" />
     <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="6.0.0" />
     <PackageReference Include="Microsoft.Extensions.Http" Version="6.0.0" />
+    <PackageReference Include="Mono.Posix.NETStandard" Version="1.0.0" />
```
Comment (dylan-smith) [inline src/gei/gei.csproj:13]: isn't this needed in the ado2gh.csproj also?

Comment (tspascoal) [reply src/gei/gei.csproj:13]: It should, at least if `ado2gh` was setting the executable bit.

Fixed in [37d14c581](https://github.com/github/gh-gei/pull/253/commits/37d14c581a1c2cbb1076efc89f48c877aa5c7a8f)

**Comment by dylan-smith** [comment]
you'll need to run `dotnet format` and commit the changes to make this pass CI.

**Comment by tspascoal** [comment]
> you'll need to run `dotnet format` and commit the changes to make this pass CI.

@dylan-smith done [18a99](https://github.com/github/gh-gei/pull/253/commits/18a9998b243b4bf797889f872dba2268c720c04c)

**Pull request merged**